cpp/bench/ann/CMakeLists.txt

```diff
@@ -27,14 +27,27 @@ option(RAFT_ANN_BENCH_SINGLE_EXE
        "Make a single executable with benchmark as shared library modules" OFF
 )
 
+# ##################################################################################################
+# * Process options ----------------------------------------------------------
+
 find_package(Threads REQUIRED)
 
-# Disable faiss benchmarks on CUDA 12 since faiss is not yet CUDA 12-enabled.
-# https://github.com/rapidsai/raft/issues/1627
-if(${CMAKE_CUDA_COMPILER_VERSION} VERSION_GREATER_EQUAL 12.0.0)
+if(CPU_ONLY)
   set(RAFT_ANN_BENCH_USE_FAISS_BFKNN OFF)
   set(RAFT_ANN_BENCH_USE_FAISS_IVF_FLAT OFF)
   set(RAFT_ANN_BENCH_USE_FAISS_IVF_PQ OFF)
+  set(RAFT_ANN_BENCH_USE_RAFT_IVF_FLAT OFF)
+  set(RAFT_ANN_BENCH_USE_RAFT_IVF_PQ OFF)
+  set(RAFT_ANN_BENCH_USE_RAFT_CAGRA OFF)
+  set(RAFT_ANN_BENCH_USE_GGNN OFF)
+else()
+  # Disable faiss benchmarks on CUDA 12 since faiss is not yet CUDA 12-enabled.
+  # https://github.com/rapidsai/raft/issues/1627
+  if(${CMAKE_CUDA_COMPILER_VERSION} VERSION_GREATER_EQUAL 12.0.0)
+    set(RAFT_ANN_BENCH_USE_FAISS_BFKNN OFF)
+    set(RAFT_ANN_BENCH_USE_FAISS_IVF_FLAT OFF)
+    set(RAFT_ANN_BENCH_USE_FAISS_IVF_PQ OFF)
+  endif()
 endif()
```

Comment:
Extending on my suggestion to add `CPU_ONLY`-like option to `ConfigureAnnBench` function:
besides defining the same-named preprocessor definition, we could use this option to disable the target within the function `ConfigureAnnBench`. This way we can avoid verbose repetition of disabling/enabling the targets here.